sklearn/utils/_array_api.py

```diff
@@ -74,6 +89,13 @@ def _check_array_api_dispatch(array_api_dispatch):
                 "array_api_compat is required to dispatch arrays using the API"
                 " specification"
             )
+        array_api_compat_version = parse_version(array_api_compat.__version__)
+        min_array_api_compat_version = "1.5.1"
```

Comment:
Let's also add this requirement to `sklearn/_min_dependencies.py` using the "tests" scope (as we do for other soft dependencies such as pandas, polars or matplotlib).

Then this line can be updated by looking up the version number from `sklearn._min_dependencies.dependent_packages`.